shipyard/src/main/java/org/jclouds/shipyard/domain/accounts/CreateAccount.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.shipyard.domain.accounts;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import org.jclouds.json.SerializedNames;
+import org.jclouds.shipyard.domain.roles.RoleInfo;
+
+import com.google.auto.value.AutoValue;
+
+@AutoValue
+public abstract class CreateAccount {
+
+   public abstract String username();
+
+   public abstract String password();
+   
+   public abstract RoleInfo role();
+   
+   CreateAccount() {
+   }
+
+   @SerializedNames({ "username", "password", "role"})
+   public static CreateAccount create(String username, String password, RoleInfo role) {
+      checkArgument(username.length() > 0, "username must not be empty string");
+      checkArgument(password.length() > 0, "password must not be empty string");
+      if (role == null) role = RoleInfo.create("user", null);
+      return new AutoValue_CreateAccount(username, password, role);
+   }
+   
+   @SerializedNames({ "username", "password", "role"})
```

Comment:
That appears to be required which is why I left it in ;) Removing it causes jclouds to complain about it not being there